File: euler001-proc.php
#!/usr/bin/env php
<?php
/** Problem 1: Multiples of 3 and 5
 * https://projecteuler.net/problem=1
 *
 * Functional is cool, procedural is also nice.
 */

function euler001(int $bound): int
{
    $sum = 0;
    for ($i = 0; $i < $bound; $i++) {
        if ($i % 3 == 0 || $i % 5 == 0) {
            $sum += $i;
        }
    }
    return $sum;
}

// Main
echo euler001(intval(trim(file_get_contents('php://STDIN')))) . PHP_EOL;

File: euler001-proc.py
#!/usr/bin/env python3

# Problem 1: Multiples of 3 and 5
# https://projecteuler.net/problem=1

import sys

def parse_input(lines):
    return int(lines[0].strip())

def euler001(bound):
    sum = 0
    for i in range(0, bound):
        if i % 3 == 0 or i % 5 == 0:
            sum += i
    return sum

if __name__ == "__main__":
    print(euler001(parse_input(sys.stdin.readlines())))
